### ISSUE TITLE
On the function eval_ score

### ISSUE THREAD

**Issue by gedizxc** [d-ailin/GDN#46]
Hello, friend. Thank you for your wonderful work in open source. I'm a novice programmer. I'd like to ask you how to write code in (eval_scores function). Some of the variables seem difficult to find in the article, such as th_ Steps, fmeas, I am very confused. I look forward to your help. Thank you！
![image](https://user-images.githubusercontent.com/29941974/190856151-03713558-b07b-4ac1-ba6b-7f8b0f1100e2.png)

**Comment by d-ailin** [comment]
Hi, this function is the evaluation function and computing the F1 score under different thresholds. The thresholds are selected by traversing th_steps times, and fmeas is a list storing the F1 score result for each threshold.

**Comment by gedizxc** [comment]
Thank you for your reply. I still wonder that the process of threshold selection I understand in paper is:
Step 1: Calculate the deviation fraction a of all sensors at a time in the verification set and normalize it
Step 2: Select the maximum sensor deviation fraction as the deviation fraction A at this moment
Step 3: Find out that the maximum A in the validation set is our threshold
Then why do you still need to use "th_ Steps "is it related to the" extreme value theory "you mentioned in the paper?

**Comment by d-ailin** [comment]
Yes, your understanding is right. The evaluation in the paper is provided at: https://github.com/d-ailin/GDN/blob/9853899da860682669a134e4af315d036aab4eca/evaluate.py#L99 The eval_scores function you mentioned is invoked from the other evaluation function: https://github.com/d-ailin/GDN/blob/9853899da860682669a134e4af315d036aab4eca/evaluate.py#L129 As our repo is also provided for custom data, we have provided two evaluation methods here, as some papers might use best F1-score as performance metrics. You could switch these evaluations by setting the parameter 'report' as  'best' or 'val'.
